ios/brave-ios/Sources/BraveVPN/Components/Region/BraveVPNRegionDetailsView.swift

```diff
@@ -21,73 +18,62 @@ struct BraveRegionDetailsView: View {
   @State
   private var isShowingChangeRegionAlert = false
 
-  @ObservedObject
-  private var cityRegionDetail: VPNCityRegionDetail
+  @StateObject
+  private var cityRegionDetail = VPNCityRegionDetail()
 
   @State
   private var regionModificationTimer: Timer?
 
-  @State
+  private var countryRegion: GRDRegion?
+
   private var cityRegions: [GRDRegion]
 
   public init(
     countryRegion: GRDRegion?,
-    with cityRegions: [GRDRegion] = [],
-    isAutoSelectEnabled: Bool = true
+    with cityRegions: [GRDRegion] = []
   ) {
-    self.isAutoSelectEnabled = isAutoSelectEnabled
+    self.countryRegion = countryRegion
     self.cityRegions = cityRegions
-
-    var regions: [BraveVPNCityRegion] = []
-
-    for cityRegion in cityRegions {
-      regions.append(
-        BraveVPNCityRegion(displayName: cityRegion.displayName, regionName: cityRegion.regionName)
-      )
-    }
-
-    cityRegionDetail = VPNCityRegionDetail(
-      countryName: countryRegion?.country ?? "",
-      countryISOCode: countryRegion?.countryISOCode ?? "",
-      cityRegions: regions
-    )
   }
 
   var body: some View {
-    ZStack {
-      List {
-        Section(header: Text(Strings.VPN.availableServerTitle)) {
-          ForEach(cityRegionDetail.cityRegions) { server in
-            cityRegionItem(at: 0, region: server)
-          }
+    List {
+      Section(header: Text(Strings.VPN.availableServerTitle)) {
+        ForEach(cityRegionDetail.cityRegions) { server in
+          cityRegionItem(region: server)
         }
-        .listRowBackground(Color(braveSystemName: .containerBackgroundMobile))
-      }
-      .opacity(isLoading ? 0.5 : 1.0)
-
-      if isLoading {
-        BraveVPNRegionLoadingIndicatorView()
-          .transition(.opacity)
-          .zIndex(1)
       }
+      .listRowBackground(Color(braveSystemName: .containerBackgroundMobile))
     }
+    .opacity(isLoading ? 0.5 : 1.0)
+    .overlay(
+      Group {
+        if isLoading {
```

Comment:
just fyi you can use the view builder version of `overlay` and remove the inner Group:

```swift
.overlay {
  if isLoading {
    ...
  }
}
```

Reply:
Nice changes are made with https://github.com/brave/brave-core/pull/24198/commits/e62c725010a6192dd0df24c1c91e3d07eb3f52ee